tools/windows/DatadogAgentInstaller/CustomActions/InstallOciPackages.cs

```diff
@@ -127,41 +131,39 @@ private ActionResult InstallPackages()
             }
         }
 
-        private ActionResult RollbackState()
-        {
-            _rollbackDataStore.Load();
-            _rollbackDataStore.Restore();
-            return ActionResult.Success;
-        }
-
-        private bool IsPackageInstalled(string library)
+        private ActionResult PurgePackages()
         {
-            var packageName = PackageName(library);
-            using (var proc = _session.RunCommand(_installerExecutable, $"is-installed {packageName}", InstallerEnvironmentVariables()))
+            if (!ShouldPurge())
             {
-                if (proc.ExitCode == 10)
-                {
-                    return false;
-                }
-
-                if (proc.ExitCode != 0)
+                _session.Log("Skipping purge as PURGE is not set to 1");
+                return ActionResult.Success;
+            }
+            try
+            {
+                _session.Log("Running datadog-installer purge");
+                var env = PurgeEnvironmentVariables();
+                using (var proc = _session.RunCommand(_installerExecutable, "purge", env))
                 {
-                    throw new Exception($"'datadog-installer is-installed {packageName}' failed with exit code: {proc.ExitCode}");
+                    if (proc.ExitCode != 0)
+                    {
+                        _session.Log($"'datadog-installer purge' failed with exit code: {proc.ExitCode}");
+                        return ActionResult.Failure;
```

Comment:
I think it is okay to fail the uninstall here.
If purge fails, then it is easier to cleanup the failure if the `datadog-installer` binary is still available on disk.
If we continue uninstall and remove everything we can, then I think next steps are less obvious.
If there is a bug/issue that's blocking uninstall, then we can set `PURGE=0` to skip the issue and complete the uninstall.
(add as comments please)